aws/resource_aws_cognito_user_pool.go

```diff
@@ -0,0 +1,902 @@
+package aws
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsCognitoUserPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoUserPoolCreate,
+		Read:   resourceAwsCognitoUserPoolRead,
+		Update: resourceAwsCognitoUserPoolUpdate,
+		Delete: resourceAwsCognitoUserPoolDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		// https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateUserPool.html
+		Schema: map[string]*schema.Schema{
+			"admin_create_user_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
```

Comment:
Just out of curiosity what's the value that comes back from the API if not defined?

Reply:
Only LambdaConfig is empty, fixing it!

Reply:
The structure gets back with a child having a default parameter

Reply:
Fair enough - as long as `Computed` is justified 🤷‍♂️